vsts/pipelines/templates/_buildTemplate.yml

```diff
@@ -29,6 +29,14 @@ steps:
     startsWith(variables['Build.SourceBranch'],'refs/heads/patch/'),
     startsWith(variables['Build.SourceBranch'],'refs/heads/exp/')))
 
+- script: |
+    sudo rm -rf /usr/share/dotnet
+    sudo rm -rf /opt/ghc
+    sudo rm -rf "/usr/local/share/boost"
+    sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+    docker images && docker system prune -fa && df -h && echo
+  displayName: 'clean docker images'
```

Comment:
Will it be faster if not doing it. Sometimes if it's cached, then build build images step will be faster?

Reply:
Such as triggering to re-run failed job directly from the pipeline, I don't know if that's an example

Reply:
unfortunately we cant ... first we dont know which agent we are gonna get .. these are auto managed by ado team .. so they recycle these agents .. secondly there is space limitation .. only 10-12 gb available ..we are sacrificing build speed here over build security @qianz2

Reply:
I see, thanks for clarifying, then I agree we should clean docker images.